File: 3451-string-compression-iii/3451-string-compression-iii.go
func compressedString(word string) string {
	l, h, c := 0, 1, 1

	var str strings.Builder

	for h < len(word) {
		if (word[l] == word[h]) && (c < 9) {
			c++
		} else {
			str.WriteString(strconv.Itoa(c))
			str.WriteByte(word[l])
			c = 1
			l = h
		}

		h++
	}

	str.WriteString(strconv.Itoa(c))
	str.WriteByte(word[l])

	return str.String()
}

File: 3451-string-compression-iii/3451-string-compression-iii.java
class Solution {
    public String compressedString(String word) {
        int l = 0, h = 1,c = 1;
        StringBuilder sb = new StringBuilder();

        while(h < word.length()){
          if(word.charAt(l) == word.charAt(h) && (c < 9)){
            c++;
          }else{
            sb.append(c);
            sb.append(word.charAt(l));

            c = 1;
            l = h;
          }

          h++;
        }

        sb.append(c);
        sb.append(word.charAt(l));

        return sb.toString();
    }
}